.github/workflows/weekly-tests.yml

```diff
@@ -20,17 +20,22 @@ jobs:
         - name: Python 3.12 (macOS)
           os: macos-latest
           python: '3.12'
-          toxenv: py312-all
+          toxenv: py312-pins-all-doctest
```

Comment:
This PR makes it so that doctests in PRs are only run in Python 3.12 with pinned dependencies.  This has the features that:
 - When there is a breaking change in fundamental constants or how arrays or quantities are displayed, we won't have to try to make our docstrings compatible with both the older and newer versions of our dependencies.
 - The code coverage metric will reflect how much the test suite itself covers lines of code, without contributions from doctests. This will make it less likely that we'll forget to include tests for new functionality.